### PR TITLE
Bug 1104139 - [Contacts] Refactor highlight so we can avoid using regexp

### PR DESCRIPTION
**Pull request by adrm**

### CODE REVIEW

**Comment by try-server-hook** [comment]
[ADLR-es Adrián de la Rosa (ADLR-es)](https://github.com/ADLR-es) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=395bdff65308)

File: apps/communications/contacts/js/contacts.js [hunk 1]
```diff
@@ -913,7 +913,10 @@ var Contacts = (function() {
     views: {
       Settings: loadFacebook,
       Details: loadFacebook,
-      Form: loadFacebook
+      Form: loadFacebook,
+      Search: function(callback) {
```
Comment (jmcanterafonseca) [inline apps/communications/contacts/js/contacts.js]: did you address Cristian's comment?

File: apps/communications/contacts/test/marionette/search_test.js [hunk 2]
```diff
@@ -16,37 +16,44 @@ marionette('Contacts > Search', function() {
...
       client.helper.waitForElement(selectors.searchInput).sendKeys('a');
       client.helper.waitForElement(selectors.searchResultFirst).text(
         function(error, textContent) {
           assert.equal('A%%&', textContent);
       });
     });
-  });
 
+    test('Search gets updated if contact changes', function() {
+      subject.addContact();
```
Comment (jmcanterafonseca) [inline apps/communications/contacts/test/marionette/search_test.js]: probably we can use contacts.createMozContact() as a safer way to create the initial contact data needed for this test

File: apps/communications/contacts/test/marionette/search_test.js [hunk 3]
```diff
@@ -16,37 +16,44 @@ marionette('Contacts > Search', function() {
   });
 
   suite('Search Mode', function() {
-    test.skip('Can enter and exit search mode', function() {
-
+    test('Can enter and exit search mode', function() {
       subject.addContact();
```
Comment (jmcanterafonseca) [inline apps/communications/contacts/test/marionette/search_test.js]: we could use createMozContact. I think it makes the test simpler and safer

File: apps/communications/contacts/test/marionette/search_test.js [hunk 4]
```diff
@@ -16,37 +16,44 @@ marionette('Contacts > Search', function() {
...
+      assert.ok(listView.displayed(), 'List view was not shown');
     });
 
-    test.skip('Search text is displayed correctly', function() {
+    test('Search text is displayed correctly', function() {
       var details = details || {
         givenName: 'A%%&'
       };
 
       subject.addContact(details);
```
Comment (jmcanterafonseca) [inline apps/communications/contacts/test/marionette/search_test.js]: same comment as above

File: apps/communications/contacts/test/unit/search_test.js [hunk 5]
```diff
@@ -147,8 +148,9 @@ suite('Search mode', function() {
       searchBox.value = contact.givenName[0][0];
       contacts.Search.search(function search_finished() {
         done(function() {
-          assert.equal(contact.givenName[0],
-                                searchList.querySelector('strong').textContent);
+          var query = 'section#groups-list-search .contact-item .contact-text';
```
Comment (jmcanterafonseca) [inline apps/communications/contacts/test/unit/search_test.js]: nit: selector

File: apps/communications/contacts/js/contacts.js (hunk 1 above)
Comment (ADLR-es) [inline apps/communications/contacts/js/contacts.js]: We talked about it and decided was better to not apply this approach here.

**Comment by try-server-hook** [comment]
[ADLR-es Adrián de la Rosa (ADLR-es)](https://github.com/ADLR-es) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=50343e64db56)

**Comment by try-server-hook** [comment]
[ADLR-es Adrián de la Rosa (ADLR-es)](https://github.com/ADLR-es) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=b762e0f17ba7)

**Comment by try-server-hook** [comment]
[ADLR-es Adrián de la Rosa (ADLR-es)](https://github.com/ADLR-es) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=4b9022195346)

**Pull request merged**